libutils/known_dirs.c

```diff
@@ -134,6 +135,21 @@ const char *GetWorkDir(void)
     return workdir == NULL ? GetDefaultWorkDir() : workdir;
 }
 
+const char *GetModuleDir(void)
+{
+    const char *module_dir = getenv("CFENGINE_TEST_OVERRIDE_MODULE_DIR");
+
+    if (module_dir == NULL) 
+    {
+        const char* workdir = GetWorkDir(); 
+        char *default_module_dir = strdup(workdir);
+        PathAppend(default_module_dir, PATH_MAX, "modules", '/');
```

Comment:
The `strdup()` function will allocate `strlen()` bytes to the heap-allocated buffer. But you are telling the `PathAppend()` that the buffer is `PATH_MAX` bytes. Hence, it will write outside the bounds of the buffer, causing undefined behavior. Instead make a buffer `PATH_MAX` bytes large and copy the first string into it before calling `PathAppend()`

Reply:
Also use FILE_SEPARATOR macro to make it platform independent. On windows backslash is used as path separator

Reply:
Maybe `snprintf()` is an easier function to use here?